src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -178,5 +200,51 @@ private static unsafe void Pad0(byte[] searchPattern, byte[] patternToReplace, b
                 }
             }
         }
+
+        /// <summary>
+        /// If the apphost file is a windows PE file (checked by looking at the first few bytes)
+        /// this method will set its subsystem to GUI.
+        /// </summary>
+        /// <param name="accessor">The memory accessor which has the apphost file opened.</param>
+        /// <param name="appHostSourcePath">The path to the source apphost.</param>
+        private static unsafe void SetWindowsGraphicalUserInterfaceBit(
+            MemoryMappedViewAccessor accessor,
+            string appHostSourcePath)
+        {
+            byte* pointer = null;
+
+            try
+            {
+                accessor.SafeMemoryMappedViewHandle.AcquirePointer(ref pointer);
+                byte* bytes = pointer + accessor.PointerOffset;
```

Comment:
Should we have a bounds check that on the view handle that it is at least `0x99` bytes in length?

Reply:
Good point - will do. Will reuse the exception message that it's not a valid Windows PE file (since it wouldn't be :-))